### DISCUSSION TITLE
Data consumption

### DISCUSSION THREAD

**Discussion by ScreamZ** [caching-tools/next-shared-cache#786] [General]
Hi @better-salmon 
> https://github.com/caching-tools/next-shared-cache/wiki/Understanding-Implicit-Tags-and-Revalidation-in-Next.js#revalidation-on-get-operation

No sure why we wait for the next Get to invalidate ?

I'm using upstash redis and as I'm paying for 20 cents for 100k commands , I'm wondering why i need to HMGET on every GET, Why just no call 

```ts
for (const timeString in revalidationTimes) {
        if (timeString && Number.parseInt(timeString, 10) > cacheValue.lastModified) {
          await redis.unlink(prefix + key);

          return null;
        }
```

directly in `revalidateTag`, as the cache will try to access data, it'll miss and this will be ok ?
Atm we are spending a HMGET on every request to `get`, I'm currently hitting at 200k request per week for each HMGET, and the app is still in beta

So the get command is doing at least one `GET` and one `HMGET` which could be done in revalidateTag ? Using Redis and TTL key, this shouldn't be mandatory as key are cleaned automatically and you don't need to check for stale content in get, like you could do in other provider that doesn't handle auto-clean ?

My connection is a pure copy of redis string implementation, but for serverless using upstash redis SDK.

**Comment by ScreamZ** [comment]
@better-salmon Any idea, would save me a lot ?

**Comment by better-salmon** [comment]
@ScreamZ hello! I'm currently working on this

**Reply by ScreamZ** [reply]
Hello @better-salmon any news on that ?
